### PR TITLE
Fix EZP-25516: Update REST spec with role draft features, fix inconsistencies

### PR DESCRIPTION
**Pull request by glye**
> Fixes https://jira.ez.no/browse/EZP-25516
> Status: Ready for review

Also remove draftByRoleId and policiesByRoleDraft routes, let the Role REST controller deal with role IDs as if there were no draft IDs.

### CODE REVIEW

**Comment by glye** [comment]
@bdunogier So far what I did is remove the routes that are not needed if we don't deal with role draft IDs, adapted the REST controller so that it will accept both role IDs and draft IDs, and deprecated the controller methods that no longer have any routes. That way 1) REST users don't need to care about draft IDs, and 2) we don't need to modify the API. It seems to work, Travis has other issues.

**Comment by glye** [comment]
This should be viewed in relation to the changes in https://github.com/ezsystems/ezpublish-kernel/commit/588d717f1ac319f853bf5ef18cb18ea5a9cc898a since it reverses some of those changes, and/or view both as a whole, e.g.:
`git diff 136a75fb2c9dedced0a15b178a6194a01e432de7..b9341cee1d032d9f00fc03066fe004d4bae4ef8d doc/specifications/rest/REST-API-V2.rst eZ/Bundle/EzPublishRestBundle/Resources/config/routing.yml eZ/Bundle/EzPublishRestBundle/Tests/Functional/RoleTest.php eZ/Publish/Core/REST/Server/Controller/Role.php`

**Comment by glye** [comment]
Review ping @bdunogier @andrerom

**Comment by andrerom** [comment]
+1 I think :)

File: doc/specifications/rest/REST-API-V2.rst [hunk 1]
```diff
@@ -5254,11 +5254,11 @@ XML Example
...
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Create Role
-```````````
+Create Role / Role Draft
+````````````````````````
 :Resource: /user/roles
 :Method: POST
-:Description: Creates a new role
+:Description: Creates a new role, or role draft if you set the POST parameter publish=false.
```
Comment (bdunogier) [inline doc/specifications/rest/REST-API-V2.rst]: In `Create Content Type`, this is formalized as follows:

```
:Description: Creates a new content type draft in the given content type group
:Parameters: :publish: (default false) If true the content type is published after creating
```

Comment (glye) [inline doc/specifications/rest/REST-API-V2.rst]: Nice, done.

**Comment by bdunogier** [comment]
Looks good. But please move the commits that change code to their own commit, and to their own issue (the one about updating the spec doesn't help explaining why we did this).

**Comment by glye** [comment]
@bdunogier If I separate spec from code, the first PR would be invalid by itself. Alternatively, I can do one PR with only the spec changes from EZP-24830, and then one PR with the code changes here *and* related spec changes (which would revert changes in the first PR). Or, I can just add the missing description to EZP-25516 :)

**Comment by bdunogier** [comment]
Update the desc then :-)

**Comment by glye** [comment]
Desc added.

**Comment by glye** [comment]
Gentle reminder ping @bdunogier

**Comment by bdunogier** [comment]
+1, with QA testing the modified/added features. Can you make sure you list what has changed in the JIRA issue so that the perimeter is clear @glye ?

**Comment by glye** [comment]
@bdunogier Ack, thanks.

**Pull request merged**